packages/core/contracts/financial-templates/perpetual-multiparty/PerpetualPositionManager.sol

```diff
@@ -160,8 +156,15 @@ contract PerpetualPositionManager is FeePayer, FundingRateApplier, Administratee
         require(_getIdentifierWhitelist().isIdentifierSupported(_priceIdentifier), "Unsupported price identifier");
 
         withdrawalLiveness = _withdrawalLiveness;
-        TokenFactory tf = TokenFactory(_tokenFactoryAddress);
-        tokenCurrency = tf.createToken(_syntheticName, _syntheticSymbol, 18);
+        tokenCurrency = ExpandedIERC20ExclusiveMinter(_tokenAddress);
+
+        // Check that tokenCurrency has an Exclusive, not Shared, Minter role.
+        // This important because this contract does not create the synthetic token, therefore
+        // it does not know to whom the original token creator has granted minter privileges.
+        require(tokenCurrency.isMinterExclusive(), "Token minter role not exclusive");
+
+        // TODO: Should this contract also check that it holds the owner role of the SyntheticToken?
```

Comment:
Adding this check would add 150 bytes

Reply:
Actually, just realized that its impossible that this contract is constructed with the owner role of the synthetic token contract. It must be given the owner role after synthetic creation